apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -238,26 +255,43 @@ export class CodeEditorViewComponent implements OnDestroy {
           // Remove link between launch project and terminals
           await this.webContainerService.loadProject(project.files, project.commands, project.cwd);
         }
-        await this.loadNewProject();
-        this.cwd$.next(project?.cwd || '');
+        this.loadNewProject();
+        this.cwd$.next(project.cwd);
       });
     });
-    this.form.controls.code.valueChanges.pipe(
-      distinctUntilChanged(),
-      skip(1),
-      debounceTime(300),
+    this.forceReload.subscribe(async () => {
+      await this.cleanAllModelsFromMonaco();
+      await this.loadAllProjectFilesToMonaco();
+    });
+    merge(
+      this.forceSave.pipe(map(() => this.form.value.code)),
+      this.form.controls.code.valueChanges.pipe(
+        distinctUntilChanged(),
+        skip(1),
+        debounceTime(1000)
+      )
+    ).pipe(
       filter((text): text is string => !!text),
       takeUntilDestroyed()
     ).subscribe((text: string) => {
       if (!this.project) {
         this.loggerService.error('No project found');
         return;
       }
+      if (text !== this.fileContent()) {
+        const { cwd } = this.project();
+        localStorage.setItem(cwd, JSON.stringify({
```

Comment:
Can we try updating and saving a change on the yarn.lock and verify it will not be truncated?

Reply:
In our use case we have a package-lock.json I don't know if there is a difference for the length of the file but it seems ok I didn't notice any latency